docs/rest-api-usage.md

```diff
@@ -0,0 +1,775 @@
+
+# Load kubeconfig
+k8s_client = config.new_client_from_config()
+dyn_client = DynamicClient(k8s_client)
+
+# Get the route
+v1_routes = dyn_client.resources.get(api_version='route.openshift.io/v1', kind='Route')
+route = v1_routes.get(name=f'ds-pipeline-{dsp_cr_name}', namespace=dsp_namespace)
+
+# Extract the hostname
+api_url = f"https://{route.spec.host}"
+client = Client(host=api_url)
+```
+
+### Basic Client Configuration
+
+The simplest way to create a client is by providing the API server host URL.
+
+```python
+import kfp
+from kfp import Client
+
+# For remote access with HTTPS
+client = Client(host='https://ds-pipeline-<dspa_name>.some.openshift.host.com')
+
+# With custom namespace (for multi-tenant deployments)
+client = Client(
+    host='https://ds-pipeline-<dspa_name>.some.openshift.host.com',
+    namespace='your-namespace'
+)
+```
+
+### Client with Custom Configuration
+
+You can customize the client behavior by specifying additional parameters such as timeout and retry settings.
+
+```python
+import kfp
+from kfp import Client
+
+# With custom timeout and retry settings
+client = Client(
+    host='https://ds-pipeline-<dspa_name>.some.openshift.host.com',
+    namespace='your-namespace',  # Optional: specify namespace for multi-tenant deployments
+    timeout=300,  # Optional: request timeout in seconds (default: 300)
+    retry_count=3,  # Optional: number of retries for failed requests (default: 3)
+    verify_ssl=True,  # Optional: verify SSL certificates (default: True)
+    ssl_ca_cert='/path/to/ca-cert.pem'  # Optional: path to CA certificate file
+)
+```
+
+## Core Operations
+
+This section covers the core operations for working with Data Science Pipelines, including managing pipelines, experiments, runs, and recurring runs.
+
+### Working with Pipelines
+
+Pipelines are the core building blocks of Data Science Pipelines. This section covers how to upload, list, retrieve, and delete pipelines using the REST API client.
+
+### Uploading a Pipeline
+
+Upload pipelines from local files or remote URLs to make them available for execution.
+
+```python
+# Upload from a local file
+pipeline = client.upload_pipeline(
+    pipeline_package_path='path/to/pipeline.yaml',
+    pipeline_name='my-pipeline',
+    description='My sample pipeline'
+)
+
+# Upload from a URL
+pipeline = client.upload_pipeline_from_url(
+    pipeline_url='https://github.com/example/pipeline.yaml',
+    pipeline_name='remote-pipeline'
+)
+```
+
+### Listing Pipelines
+
+Retrieve a list of pipelines with support for pagination and filtering.
+
+```python
+# List all pipelines
+pipelines = client.list_pipelines()
+print(f"Found {pipelines.total_size} pipelines")
+
+for pipeline in pipelines.pipelines:
+    print(f"Pipeline: {pipeline.display_name} (ID: {pipeline.pipeline_id})")
+
+# List with pagination
+pipelines = client.list_pipelines(page_size=10, page_token=None)
+
+# List with filtering
+pipelines = client.list_pipelines(filter='name="my-pipeline"')
```

Comment:
Could you update it use JSON-serialized Filter proto, as it's not accepting string expression.
https://github.com/opendatahub-io/data-science-pipelines/blob/master/sdk/python/kfp/client/client.py#L655